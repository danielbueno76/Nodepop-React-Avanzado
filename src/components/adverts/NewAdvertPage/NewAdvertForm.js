import React from "react";
import { FormField, Radio, Select, Form, FormButton } from "../../shared";
import { BUY, SELL } from "../../../utils/utils";
import { useSelector, useDispatch } from "react-redux";
import { advertsTagsAction } from "../../../store/actions";
import { getTags } from "../../../store/selectors";

const NewAdvertForm = ({ onSubmit }) => {
  const dispatch = useDispatch();

  const allTags = useSelector(getTags);
  React.useEffect(() => {
    dispatch(advertsTagsAction());
  }, [dispatch]);

  return (
    <Form
      initialValue={{
        name: "",
        price: "",
        sale: null,
        tags: [],
      }}
      onSubmit={onSubmit}
    >
      <FormField type="text" name="name" autofocus />
      <FormField type="number" name="price" />
      <Radio name="sale" arrayValues={[BUY, SELL]} />
      <Select name="tags" allTags={allTags} />
      <FormField type="file" name="photo" accept="image/*" />
      <FormButton type="submit" variant="primary" id="form-submit">
        Publish
      </FormButton>
    </Form>
  );
};

export default NewAdvertForm;
