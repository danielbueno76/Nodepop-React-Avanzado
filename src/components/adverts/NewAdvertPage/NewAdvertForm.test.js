import React from "react";
import { shallow } from "enzyme";
import NewAdvertForm from "./NewAdvertForm";

jest.mock("react-redux");

describe("NewAdvertForm", () => {
  const props = {
    onSubmit: jest.fn(),
  };

  const render = () => shallow(<NewAdvertForm {...props} />);
  const wrapper = render();

  test("should render", () => {
    expect(wrapper.exists()).toBe(true);
  });

  test("snapshot testing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test("should find 3 FormField", () => {
    expect(wrapper.find("FormField")).toHaveLength(3);
  });

  test("should call event onSubmit", () => {
    const form = wrapper.find("Form");
    form.simulate("submit", { preventDefault: jest.fn });

    expect(props.onSubmit).toBeCalled();
  });
});
