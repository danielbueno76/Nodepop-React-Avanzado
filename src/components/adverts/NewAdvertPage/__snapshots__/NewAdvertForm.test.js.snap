// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewAdvertForm snapshot testing 1`] = `
<Form
  initialValue={
    Object {
      "name": "",
      "price": "",
      "sale": null,
      "tags": Array [],
    }
  }
  onSubmit={[MockFunction]}
>
  <FormField
    autofocus={true}
    name="name"
    type="text"
  />
  <FormField
    name="price"
    type="number"
  />
  <Radio
    arrayValues={
      Array [
        "Buy",
        "Sell",
      ]
    }
    name="sale"
  />
  <Select
    name="tags"
  />
  <FormField
    accept="image/*"
    name="photo"
    type="file"
  />
  <FormButton
    id="form-submit"
    type="submit"
    variant="primary"
  >
    Publish
  </FormButton>
</Form>
`;
